lib/logstash/inputs/xmpp.rb

```diff
@@ -32,26 +36,32 @@ class LogStash::Inputs::Xmpp < LogStash::Inputs::Base
   config :debug, :validate => :boolean, :default => false, :deprecated => "Use the logstash --debug flag for this instead."
 
   public
-  def register
-    require 'xmpp4r' # xmpp4r gem
-    Jabber::debug = true if @debug || @logger.debug?
 
+  def initialize(config)
+    super
     @client = Jabber::Client.new(Jabber::JID.new(@user))
+    @muc_clients = []
+  end
+
+  def register
+    Jabber::debug = true if @debug || @logger.debug?
     @client.connect(@host) # it is ok if host is nil
     @client.auth(@password.value)
     @client.send(Jabber::Presence.new.set_type(:available))
-
-    # load the MUC Client if we are joining rooms.
-    require 'xmpp4r/muc/helper/simplemucclient' if @rooms && !@rooms.empty?
   end # def register
 
-  public
+  # for testing access
+  attr_reader :client, :muc_clients
```

Comment:
My thinking: as its used in methods from register downwards it should be above `def register`.

We don't really enforce the classic ruby file layout and for plugins I think the config DSL should standout at the top. My 2c.